src/date.ts

```diff
@@ -0,0 +1,175 @@
+import 'globalize';
+import 'globalize/dist/globalize/date';
+import 'globalize/dist/globalize/relative-time';
+import getGlobalize from './util/globalize';
+
+export type DateLength = 'short' | 'medium' | 'long' | 'full';
+export type RelativeTimeLength = 'short' | 'narrow';
+
+export type DateFormatterOptions = {
+	/**
+	 * String value indicating a skeleton, eg. { skeleton: "GyMMMd" }.
+	 * Skeleton provides a more flexible formatting mechanism than the predefined list full, long, medium, or short represented by date, time, or datetime.
+	 * Instead, they are an open-ended list of patterns containing only date field information, and in a canonical order.
+	 */
+	skeleton?: string;
+
+	/**
+	 * One of the following String values: full, long, medium, or short, eg. { date: "full" }.
+	 */
+	date?: DateLength;
+
+	/**
+	 * One of the following String values: full, long, medium, or short, eg. { time: "full" }.
+	 */
+	time?: DateLength;
+
+	/**
+	 * One of the following String values: full, long, medium, or short, eg. { datetime: "full" }
+	 */
+	datetime?: DateLength;
+}
+
+export type RelativeTimeFormatterOptions = {
+	/**
+	 * eg. "short" or "narrow". Or falsy for default long form
+	 */
+	form?: RelativeTimeLength;
+}
+
+/**
+ * Format a date according to the specified options for the specified or current locale.
+ *
+ * @param value
+ * The date to format.
+ *
+ * @param options
+ * An optional object of formatting options.
+ *
+ * @param locale
+ * An optional locale. Defaults to the root locale.
+ *
+ * @return
+ * The formatted date string.
+ */
+export function formatDate(value: Date, options?: DateFormatterOptions, locale?: string): string;
+export function formatDate(value: Date, locale?: string): string;
+export function formatDate(value: Date, optionsOrLocale?: DateFormatterOptions | string, locale?: string): string {
+	const options = typeof optionsOrLocale === 'string' ? undefined : optionsOrLocale;
+	locale = typeof optionsOrLocale === 'string' ? optionsOrLocale : locale;
+	return getGlobalize(locale).formatDate(value, options);
```

Comment:
I had tried a couple different things, but eventually settled on allowing duplication of this particular pattern since anything else I had attempted felt worse. I'll see if I can make this delegator function work.